filebeat/fileset/modules.go

```diff
@@ -69,33 +69,13 @@ func newModuleRegistry(modulesPath string,
 			return nil, fmt.Errorf("error getting filesets for module %s: %v", mcfg.Module, err)
 		}
 
-		for _, filesetName := range moduleFilesets {
-			fcfg, exists := mcfg.Filesets[filesetName]
-			if !exists {
-				fcfg = &FilesetConfig{}
-			}
+		for filesetName, fcfg := range mcfg.Filesets {
 
 			fcfg, err = applyOverrides(fcfg, mcfg.Module, filesetName, overrides)
 			if err != nil {
 				return nil, fmt.Errorf("error applying overrides on fileset %s/%s: %v", mcfg.Module, filesetName, err)
 			}
 
-			if fcfg.Enabled != nil && !(*fcfg.Enabled) {
-				continue
-			}
-
-			fileset, err := New(modulesPath, filesetName, mcfg, fcfg)
-			if err != nil {
-				return nil, err
-			}
-			if err = fileset.Read(beatInfo); err != nil {
-				return nil, fmt.Errorf("error reading fileset %s/%s: %v", mcfg.Module, filesetName, err)
-			}
-			reg.registry[mcfg.Module][filesetName] = fileset
-		}
-
-		// check that no extra filesets are configured
-		for filesetName, fcfg := range mcfg.Filesets {
 			if fcfg.Enabled != nil && !(*fcfg.Enabled) {
```

Comment:
due to https://github.com/elastic/beats/pull/27762 we could have a case where `fcfg.Enabled = nil` and the fileset was not skipped.
now `fcfc` comes from defined filesets: https://github.com/elastic/beats/pull/27526/files#diff-515d023121b7ec77d64c2a0cfcfe3dc6f9b46e19166e71ac34e5f5cf74fddfe6R72, but still it can be the case that `enabled: false` is not defined.
I'd change this to `if fcfg.Enabled == nil || ( != nil && !(*fcfg.Enabled))`

what do you think @adriansr ?

Reply:
This PR maintains the current behavior of the enabled flag: If it's not defined, it defaults to `true`. So `enabled==nil || *enabled` is the true check, and `enabled!=nil && !(*enabled)` is the false check.

Reply:
got it @adriansr, but what do you think about changing this behaviour?

cc @exekias this is related to what we discussed in https://github.com/elastic/beats/pull/27612

We could address in another PR in case, just opening the discussion

Reply:
The problem with this change is that it introduces another breaking change, and will potentially break existing configurations for our users. Also it makes configuring filesets from the command line more complicated as it forces users to always include an enabled flag `-M mymodule.mydataset.enabled=true`.

Reply:
I understand your concern, it is actually a breaking change.

The problem I see in enabling by default if not explicitly disabled is that we miss the chance to enforce stricter validation on filesets where there is not a default not failing config (see #27612 ). 
In the specific case there's no option to either prevent filebeat from starting if we introduce the validation, or let it start with a misconfigured explicitly enabled fileset that will stop as input.

Also what we have as current behaviour is the possibility of filesets starting and stopping because "implicitly" enabled and not properly configured, producing noisy logs that could be confusing.

Hard choice to take, indeed